src/main/java/ether/TransactionsManager.java

```diff
@@ -0,0 +1,663 @@
+        insertTransaction(event);
+    }
+
+    /**
+     * Inserts transaction to the transactions queue.
+     * @param event
+     */
+    private void insertTransaction(MeshManager.MeshTransactionEvent event){
+        if(!transactionsQueue.offer(event)){
+            if(Settings.DEBUG_INFO){
+                System.out.println("Failed to add transaction to the queue. Probably the queue is full.");
+                System.out.println("Num of elements in the queue: " + transactionsQueue.size());
+            }
+        }
+    }
+
+    /**
+     * Processes Transactions queue in a separate managed Thread.
+     */
+    private void processTransactionsQueue(){
+        while(isRunning){
+            MeshManager.MeshTransactionEvent event;
+            try {
+                event = transactionsQueue.take();
+            } catch (InterruptedException e) {
+                if(Settings.DEBUG_INFO){
+                    System.out.println("InterruptedException: "+e);
+                }
+                continue;
+            }
+
+            //execute on a separate Thread
+            executor.execute(() -> processTransaction(event));
+        }
+    }
+
+
+    /**
+     * Process the Mesh network trnasctions from Clients-Remote Peers.
+     * @param event
+     */
+    private void processTransaction(MeshManager.MeshTransactionEvent event){
+        MeshTransaction transaction = event.transaction;
+        byte[] transactionData = new byte[0]; //TODO: transaction.data;
+
+        JSONObject jsonObject;
+        JSONParser parser = new JSONParser();
+        try {
+            jsonObject = (JSONObject) parser.parse(new String(transactionData));
+        } catch (ParseException e) {
+            if(Settings.DEBUG_INFO){
+                System.out.println("Failed to parse transaction, ParseException: "+e);
+            }
+            return;
+        }
+
+        String method = (String)jsonObject.get("method");
+        switch(method) {
+            case "stop":
+                //Dummy packet - Do nothing
+                break;
+            case EtherUtility.METHOD_GET_ALL:
+                processGetAllRequest(event.peerUuid);
+                break;
+
+            case EtherUtility.METHOD_OPEN_CLIENT_TO_SUPER_PEER:
+                processOpenClientToSuperPeerRequest(event.peerUuid, jsonObject);
+                break;
+
+            case EtherUtility.METHOD_BALANCE_MSG_SIG:
+                processBalanceMessageSignatureRequest(event.peerUuid, jsonObject);
+                break;
+
+            default:
+                if(Settings.DEBUG_INFO){
+                    System.out.println("default case in processTransaction method.");
+                }
+                break;
+        }
+    }
+
+
+    /**
+     * Processes the Get All request from Client-Remote Peer
+     * @param sourceId
+     */
+    private void processGetAllRequest(MeshID sourceId) {
```

Comment:
This is a _really_ long function - can any chunks of it be moved into their own functions?